# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/openjdk/add-missing-arch.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- jdk-jdk-23-ga/src/java.base/share/classes/jdk/internal/util/Architecture.java.vanilla	2024-11-17 15:47:25.480150781 +0100
+++ jdk-jdk-23-ga/src/java.base/share/classes/jdk/internal/util/Architecture.java	2024-11-17 15:56:17.040160408 +0100
@@ -44,6 +44,8 @@
     AARCH64(64, ByteOrder.LITTLE_ENDIAN),
     ARM(32, ByteOrder.LITTLE_ENDIAN),
+    HPPA(32, ByteOrder.BIG_ENDIAN),
     LOONGARCH64(64, ByteOrder.LITTLE_ENDIAN),
+    M68K(32, ByteOrder.BIG_ENDIAN),
     MIPSEL(32, ByteOrder.LITTLE_ENDIAN),
     MIPS64EL(64, ByteOrder.LITTLE_ENDIAN),
     OTHER(is64bit() ? 64 : 32, ByteOrder.nativeOrder()),
@@ -52,6 +53,7 @@
     PPC64LE(64, ByteOrder.LITTLE_ENDIAN),
     RISCV64(64, ByteOrder.LITTLE_ENDIAN),
     S390(64, ByteOrder.BIG_ENDIAN),
+    SH(32, ByteOrder.LITTLE_ENDIAN),
     SPARCV9(64, ByteOrder.BIG_ENDIAN),
     X86(32, ByteOrder.LITTLE_ENDIAN),
     X64(64, ByteOrder.LITTLE_ENDIAN),  // Represents AMD64 and X86_64
